bdai_ros2_wrappers/bdai_ros2_wrappers/subscription.py

```diff
@@ -78,3 +81,106 @@ def wait_for_message(
         future.cancel()
         return None
     return future.result()
+
+
+def wait_for_messages(
+    topics: typing.List,
+    mtypes: typing.List,
+    *,
+    node: typing.Optional[rclpy.node.Node] = None,
+    timeout_sec: typing.Optional[float] = None,
+    **kwargs: typing.Any,
+) -> typing.Union[None, typing.List[MessageT]]:
+    """Waits for messages to arrive at multiple topics within a given time window.
+
+    Uses message_filters.ApproximateTimeSynchronizer. This function blocks
```

Comment:
@kaiyu-zheng meta: would it make to allow the user to pass its own synchronizer classes? Furthermore, we could define function arguments such that `**kwargs` can be forwarded to the synchronizer constructor directly.